crates/volta-core/src/log.rs

```diff
@@ -0,0 +1,152 @@
+                Level::Warn => self.log_warning(record.args()),
+                Level::Debug => println!("[verbose] {}", record.args()),
+                _ => println!("{}", record.args()),
+            }
+        }
+    }
+
+    fn flush(&self) {}
+}
+
+impl Logger {
+    /// Initialize the global logger with a Logger instance
+    /// Will use the requested level of Verbosity
+    /// If set to Default, will use the environment to determine the level of verbosity
+    pub fn init(context: LogContext, verbosity: LogVerbosity) -> Result<(), SetLoggerError> {
+        let logger = Logger::new(context, verbosity);
+        log::set_max_level(logger.level);
+        log::set_boxed_logger(Box::new(logger))?;
+        Ok(())
+    }
+
+    fn new(context: LogContext, verbosity: LogVerbosity) -> Self {
+        let level = match verbosity {
+            LogVerbosity::Quiet => LevelFilter::Error,
+            LogVerbosity::Default => level_from_env(),
+            LogVerbosity::Verbose => LevelFilter::Debug,
+        };
+
+        Logger { context, level }
+    }
+
+    fn log_error<D>(&self, message: &D)
+    where
+        D: Display,
+    {
+        let prefix = match &self.context {
+            LogContext::Volta => ERROR_PREFIX,
+            LogContext::Shim => SHIM_ERROR_PREFIX,
+        };
+
+        eprintln!("{} {}", style(prefix).red().bold(), message);
+    }
+
+    fn log_warning<D>(&self, message: &D)
+    where
+        D: Display,
+    {
+        let prefix = match &self.context {
+            LogContext::Volta => WARNING_PREFIX,
+            LogContext::Shim => SHIM_WARNING_PREFIX,
+        };
+
+        println!(
+            "{}{}",
+            style(prefix).yellow().bold(),
+            wrap_content(prefix, message)
+        );
+    }
+}
+
+/// Wraps the supplied content to the terminal width, if we are in a terminal.
+/// If not, returns the content as a String
+///
+/// Note: Uses the supplied prefix to calculate the terminal width, but then removes
+/// it so that it can be styled (style characters are counted against the wrapped width)
+fn wrap_content<D>(prefix: &str, content: &D) -> String
+where
+    D: Display,
+{
+    match text_width() {
+        Some(width) => Wrapper::with_splitter(width, NoHyphenation)
+            .subsequent_indent(WRAP_INDENT)
+            .break_words(false)
+            .fill(&format!("{} {}", prefix, content))
+            .replace(prefix, ""),
+        None => format!(" {}", content),
+    }
+}
+
+/// Determines the correct logging level based on the environment
+/// If VOLTA_LOGLEVEL is set to a valid level, we use that
+/// If not, we check the current stdout to determine whether it is a TTY or not
+///     If it is a TTY, we use Info
+///     If it is NOT a TTY, we use Error as we don't want to show warnings when running as a script
+fn level_from_env() -> LevelFilter {
+    match env::var(VOLTA_LOGLEVEL).as_ref() {
+        Ok(l) if l == "off" => LevelFilter::Off,
```

Comment:
I _think_ you could write:
```rust
Ok("off") => LevelFilter::Off,
// etc
```
but maybe you tried that already.

Reply:
I did try that initially, the issue is that `env::var` returns a `Result<String, VarError>`, so it throws a type error when trying to match against an `&str` literal. And adding something like `"off".into()` then complains that it's not an Enum variant because it has to calculate something to do the match.